src/netaddress.h

```diff
@@ -107,6 +109,22 @@ class CNetAddr
         }
 
         friend class CSubNet;
+        friend class CNetAddrHash;
+};
+
+class CNetAddrHash
+{
+public:
+    size_t operator()(const CNetAddr& a) const noexcept
+    {
+        CSipHasher hasher(salt_k0, salt_k1);
+        hasher.Write(a.ip, sizeof(a.ip));
+        return (size_t)hasher.Finalize();
+    }
+
+private:
+    const uint64_t salt_k0 = GetRand(std::numeric_limits<uint64_t>::max());
+    const uint64_t salt_k1 = GetRand(std::numeric_limits<uint64_t>::max());
```

Comment:
nit: Class member naming convention?

Reply:
Right! A convention is a convention. Renamed.